fixed multiple typos inside docs (#6749)
rotki/rotki: docs/api.rst

   .. note::
      This endpoint can also be queried asynchronously by using ``"async_query": true``

>    Doing a PUT on the evm transactions endpoint will request a decoding of the given transactions and generation of decoded events. That basically entails querying the transaction receipts for each transaction hash and then decoding all events. If events are already queried and ignore_cache is true they will be deleted and required.

Reviewer: you still have the same mistake here
```suggestion
   Doing a PUT on the evm transactions endpoint will request a decoding of the given transactions and generation of decoded events. That basically entails querying the transaction receipts for each transaction hash and then decoding all events. If events are already queried and ignore_cache is true they will be deleted and re-queried.
```

   :reqjson list data[optional]: A list of data to decode. Each data entry consists of an ``"evm_chain"`` key specifying the evm chain for which to decode tx_hashes and a ``"tx_hashes"`` key which is an optional list of transaction hashes to request decoding for in that chain. If the list of transaction hashes is not passed then all transactions for that chain are decoded. Passing an empty list is not allowed.
   :reqjson bool async_query: Boolean denoting whether this is an asynchronous query or not

>    :reqjson bool ignore_cache: Boolean denoting whether to ignore the cache for this query or not. This is always false by default. If true is given then the decoded events will be deleted and required.

Reviewer: same here

.. http:patch:: /api/(version)/assets/icon/modify

   Doing a PATCH on the asset icon endpoint will refresh the icon of the given asset.

>    First, the cache of the icon of the given asset is deleted and then required from CoinGecko and saved to the filesystem.

Reviewer: This is wrong. It's not required. It's re-queried.

```suggestion
   First, the cache of the icon of the given asset is deleted and then re-queried from CoinGecko and saved to the filesystem.
```

Author: done

      {"result": true, "message": ""}

>    :statuscode 200: Icon successfully deleted and required.

Reviewer: same here

```suggestion
   :statuscode 200: Icon successfully deleted and re-queried.
```
